deploy/jenkins-ci/README.md

```diff
@@ -0,0 +1,203 @@
+
+* [bin/entrypoint.sh](bin/entrypoint.sh)
+* [Dockerfile](Dockerfile)
+* [jenkins_ref/*](jenkins_ref)
+
+are provided to create a container that can be used to test
+the Pipeline.
+
+### README files
+
+The [README.md](README.md) file is what you are reading. The files
+under the `assets` directory are used in the [README.md](README.md).
+
+## Developer setup
+
+Modifying the Pipeline is easy. You just need a text editor :smile:.
+See the [Directory contents](#directory-contents) section to have
+some understanding about which files you need to change.
+
+The "hard" part is to test the Pipeline. You need a Jenkins instance
+with the tools to correctly build both Kiali's back-end and
+front-end, and to deploy Kiali. Instead of going through all the
+steps to setup such Jenkins instance, a [preconfigured
+Docker image](https://hub.docker.com/r/edgarhz/kiali-jenkins) is
+available to start as fast as possible. Run it by invoking:
+
+```
+docker run \
+  -v /var/run/docker.sock:/var/run/docker.sock \
+  -p 8080:8080 \
+  edgarhz/kiali-jenkins
+```
+
+This command assumes that your Docker daemon's socket is in
+`/var/run/docker.sock`. The `-v` parameter is needed because
+Kiali is released as a container image, which means that Jenkins
+needs access to Docker to build and push the container. Using the
+provided command, Jenkins will re-use the Docker instance of
+your machine.
+
+Once the container is running, use your browser to access Jenkins
+at http://localhost:8080. Username and password are both `admin`.
+
+**Note:** The `edgarhz/kiali-jenkins` image is updated now and then.
+So, you may find it is old. But it can work for testing.
+
+The image is preconfigured with two jobs:
+* **kiali-release:** It is the Pipeline you will be working with.
+* **kiali-release-notifier:** It does nothing. It's there because
+  the Pipeline invokes this job. In the real setup, it
+  does some QE tasks.
+
+### Building the Jenkins image
+
+In case you prefer to build the Jenkins image, all required files
+are provided in the [/deploy/jenkins-ci](/deploy/jenkins-ci)
+directory of the repository. Inside that directory, simply run:
+
+`docker build -t kiali-jenkins .`
+
+And run the generated image:
+
+```
+docker run \
+  -v /var/run/docker.sock:/var/run/docker.sock \
+  -p 8080:8080 \
+  kiali-jenkins
+```
+
+Check the [Dockerfile](Dockerfile) to learn how Jenkins is set up.
+
+### Setup Jenkins credentials
+
+The only thing that is not setup in the preconfigured Jenkins
+image are _credentials_. I hope you can guess why :smile:.
+
+Before trying anything, you need to properly setup the credentials.
+Once you are logged into Jenkins, click _Credentials_ in the menu
+at the left, then click _(global)_ in the page that appears:
+
+![Credentials Home](assets/jk_credentials_home.png)
+
+In the _Global credentials_ page, use the _Add Credentials_ option
+at the left to add these five credentials:
+
+* **GitHub SSH keys of the "bot" account**: Used to checkout
```

Comment:
I think you can also use the "GitHub token of the bot account" instead of the SSH key.

https://github.com/settings/tokens
> Personal access tokens function like ordinary OAuth access tokens. They **can be used instead of a password for Git over HTTPS**, or can be used to authenticate to the API over Basic Authentication.

Reply:
Sounds like a good improvement. I'll consideer that on the next change of the Pipeline.